src/database/query/project.ts

```diff
@@ -401,6 +418,110 @@ const deleteChapterFromSectionInDB = async ({
   }
 };
 
+const updateUserProjectChapterInDB = async ({
+  userId,
+  projectId,
+  sectionId,
+  chapterId,
+  isCompleted,
+}: UpdateUserChapterInProjectRequestProps) => {
+  try {
+    // Find the UserProject document
+    const userProject = await UserProject.findOne({ userId, projectId });
+
+    if (!userProject) {
+      return { error: 'User project not found' };
+    }
+
+    // Find the specific section in the sections array
+    const sectionIndex = userProject.sections.findIndex(
+      (section) => section.sectionId.toString() === sectionId.toString()
+    );
+
+    if (sectionIndex === -1) {
+      return { error: 'Section not found in user project' };
+    }
+
+    // Find the specific chapter in the chapters array within the section
+    const chapterIndex = userProject.sections[sectionIndex].chapters.findIndex(
+      (chapter) => chapter.chapterId.toString() === chapterId.toString()
+    );
+
+    if (chapterIndex === -1) {
+      // If chapter is not found, add it with the given status
+      userProject.sections[sectionIndex].chapters.push({
+        chapterId: chapterId,
+        isCompleted: isCompleted,
+        createdAt: new Date(),
+        updatedAt: new Date(),
```

Comment:
These two timestamps gets updated Automatically if your DB model is having timestamp=true prop

Reply:
resolved